### ISSUE TITLE
Market Cost Pulls Sell Price

### ISSUE THREAD

**Issue by athaller** [kokarn/tarkov-tools#94]
When looking at the Hideout Crafts page -- the Cost column should indicate the buy price of items based on a filter setting (either market price or trader barter price) but is currently showing the sell price of the item on the market

TLDR;

1 bug
1 feature enhancement (filter)

Gonna scan around the src and possibly send a PR, just wanted to log a ticket prior to doing so

**Comment by athaller** [comment]
Adding notes after looking into the src a bit. The reason why it does this is it pulls the lowest24h price after making the API call. While this is good for general terms I have some thoughts on how this can handle a few different options for users

- Hideout Module
	- Calculation filter should filter the multi-dimensional array first, then filter the outter array
	- Each item should have another table to show the different price/cost options based on the following:
		- Flea Market Price
		- Trader Price
		- Avg 24h price
		- Possibly craft price?


TLDR; basically get each item in the hideout module in one array, then for each item calculate the cost for the above options (flea, trader, avg24h). Then use the parent table filter to sort which child item should be displayed in the parent item. The hideout item should still have the ability of drilling down to see the other cost options that are not the best match from the filter

Looking at the graphql data request, I believe the data is already available but the functions would need to still be added and the html markup to the table/css edits

**Comment by kokarn** [comment]
I agree on the bug that it should use the "buyFor" price, but not sure I agree with the other suggestions.

I'm a firm believer in "choices not options" eg that we as developers should make choices that best suit our users and not add a lot of settings just because.

What's the use-case for showing with & without flea prices automatically on the same page and do you have any suggestions on how that layout would look?

**Comment by athaller** [comment]
I figured it would be good to give the choice as viewing cost of buying item on market as of now vs cost of buying the items to craft said item -- so you could weigh the time aspect in with the overall cost/profit. 

TBH after creating this ticket, I was thinking if it made more sense to maybe have a component to allow users to pivot the data in a pivot table.. but that could be a complex component to introduce to the site as I expect it'd need an advanced user willing to learn the graphQL syntax to query for said data to pivot on. So I didn't suggest/recommend as I don't know how much it makes sense


In terms of how this should look. My original idea was to have a sub-table below each item showing the different choices

e.g.,

ledx --- best profit price displayed here (ideally I think this level should show what the current table's filter is on. [duration, cost, flea throughput, estimated profit, estimated profit per hour] )
--- ledx -- cost to craft
--- ledx -- cost to purchase on market via current api data
--- ledx -- avg24 hour cost

Note: the sub-data showing the different choices they have can be expandable

so if one were to sort on best ledx, they could see the best price might be to buy it on the market but could weigh having x items already


**A much more simplier way would be to just have another dropdown filter at the top and change the data based on the cost factor option (24hour, market price, craft price)**
